File: adventofcode/2024/01_1.cpp
#include <bits/stdc++.h>
#define int long long

using namespace std;

signed main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    #ifndef ONLINE_JUDGE 
        freopen("../input.txt", "r", stdin); 
        freopen("../output.txt", "w", stdout); 
    #endif
    const int LINES=1000;

    vector<int> x(LINES), y(LINES);
    for(int i=0; i<LINES; i++) {
        cin >> x[i] >> y[i];
    }
    sort(x.begin(), x.end());
    sort(y.begin(), y.end());
    int ans = 0;
    for(int i=0; i<LINES; i++) {
        ans += abs(x[i]-y[i]);
    }
    cout << ans;
    return 0;
}

File: adventofcode/2024/python/01_1.py
import sys

def main():
    sys.stdin = open("../input.txt", "r")
    sys.stdout = open("../output.txt", "w")
    
    LINES = 1000
    
    # Read input
    x = []
    y = []
    for _ in range(LINES):
        a, b = map(int, input().split())
        x.append(a)
        y.append(b)
    
    # Sort the lists
    x.sort()
    y.sort()
    
    # Calculate the answer
    ans = 0
    for i in range(LINES):
        ans += abs(x[i] - y[i])
    
    print(ans)

if __name__ == "__main__":
    main()